### ISSUE TITLE
Inverse linearity schema incompatible with CRDS

### ISSUE THREAD

**Issue by alphasentaurii** [spacetelescope/rad#292]
CRDS certify checks asdf reference files against the schema uri (or "id"), and this check has to be valid in order for a reference file submission to be successful. The reftype/dm for "inverselinearity" is looking for an asdf tag of "inverselinearity", but the schema in RAD is still using the underscore: https://github.com/spacetelescope/rad/blob/main/src/rad/resources/manifests/datamodels-1.0.yaml. Submissions for new inverselinearity reference files are not possible until the schema is revised to match the rmaps/reference files (no underscore between inverse and linearity).

https://github.com/spacetelescope/rad/blob/main/src/rad/resources/manifests/datamodels-1.0.yaml

For reference, this is how crds.certify checks the asdf tags for Roman: https://github.com/spacetelescope/crds/blob/08da10721c0e979877dc9579b4092c79f4ceee27/crds/certify/certify.py#L496

In short, if f = the open asdf file:
f['roman']['meta']['reftype'].lower() needs to match f['roman']._tag)

**Comment by WilliamJamieson** [comment]
The tag and `id` (schema uri) are independent in ASDF:

- The tag is defined here in the manifiest: https://github.com/spacetelescope/rad/blob/a9203cfdce963bd518cfe8f44185470707a38faa/src/rad/resources/manifests/datamodels-1.0.yaml#L149-L150 (note that this is associating the `tag_uri` value with the `schema_uri` (`id` in the schema).
- While the `id`/schema uri is defined here: https://github.com/spacetelescope/rad/blob/a9203cfdce963bd518cfe8f44185470707a38faa/src/rad/resources/schemas/reference_files/inverse_linearity-1.0.0.yaml#L4

Looking at the code you provided, it appears that you are having an issue with the tag value, not the `id`. So is your issue actually with the tag uri?

**Comment by alphasentaurii** [comment]
For this specific case, yes I'm referring to the tag_uri, assuming that is what is used to create the ._tag attribute for the asdf reference file. That said, I'm curious why you wouldn't want the 'id' to be consistent, knowing that the name for this reftype is "inverselinearity" everywhere else?

`maker_utils.mk_inverse_linearity(shape=(2,2,2), filepath="roman_wfi_inverselinearity_dmtest.asdf")`
`f = asdf.open("roman_wfi_inverselinearity_dmtest.asdf")`
`f['roman']['meta']['reftype'].lower() == f['roman']._tag`  _should return True_
`False`

**Comment by alphasentaurii** [comment]
Gotcha. Yes, the one thing I didn't refer to (tag_uri) is the one to update 😆  Thanks William

**Comment by WilliamJamieson** [comment]
The ASDF tag/schema uri mismatch is supported so that one can use the same schema for multiple tags. Some things serialize the same but you want different objects in/out. E.G. sine and cosine models are parameterized the same so should serialize the same, just with a different "tag". So there many uris you have to keep straight, and the URI might not match the file name (ASDF/jsonschema do not care about the name of the actual schema file). So it becomes a mess.
